index.d.ts

```diff
@@ -40,6 +44,11 @@ declare module 'apollo-datasource-mongodb' {
       options?: Options
     ): Promise<(TData | null | undefined)[]>
 
+    findByFields(
+      fields: Fields,
```

Comment:
Should this not be a partial of `TData`? :thinking: 

```suggestion
      fields: Partial<TData>,
```

Reply:
nvm, the data might not be an array, but we can in fact query by arrays.